pkg/apiserver/diagnose/diagnose.go

```diff
@@ -179,15 +179,12 @@ func (s *Service) reportHTMLHandler(c *gin.Context) {
 		c.Data(http.StatusNotFound, "text/plain", []byte("UI is not built"))
 		return
 	}
+	defer func(old string) {
+		c.Request.URL.Path = old
+	}(c.Request.URL.Path)
 
-	// Serve report html directly from assets
-	d, err := s.uiAssetFS.Asset("build/diagnoseReport.html")
-	if err != nil {
-		c.Status(http.StatusNotFound)
-		return
-	}
-
-	c.Data(http.StatusOK, "text/html; charset=utf-8", d)
+	c.Request.URL.Path = "diagnoseReport.html"
+	httpgzip.FileServer(s.uiAssetFS, httpgzip.FileServerOptions{IndexHTML: true}).ServeHTTP(c.Writer, c.Request)
```

Comment:
can we access the content directly instead of constructing a FileServer each request?

Reply:
Without the FileServer, we need to handle content-encoding by ourselves.